AdyenCard/Validators/CardNumberValidator.swift

```diff
@@ -38,12 +39,30 @@ public final class CardNumberValidator: Validator {
         self.panLength = panLength
     }
     
-    public func isValid(_ value: String) -> Bool {
-        guard isEnteredBrandSupported else { return false }
-        let minimumValidCardLength = 12
-        let isValid = value.count >= minimumValidCardLength && (!isLuhnCheckEnabled || luhnCheck(value))
+    public func validate(_ value: String) -> ValidationStatus {
+        // order of checks are important to return the correct error.
+        
+        if !isEnteredBrandSupported {
+            return .invalid(CardValidationError.cardUnsupported)
+        }
+        
+        if value.isEmpty {
+            return .invalid(CardValidationError.cardNumberEmpty)
+        }
+        
+        if value.count < 12 {
```

Comment:
This is a magic number, let's extract it to a named constant?

Reply:
good catch! 776dbfd3ef4ea752d9bdeef328c4327425240f4b